experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/master/ResourceManagerCallbackHandler.scala

```diff
@@ -60,19 +76,19 @@ class ResourceManagerCallbackHandler(appConfig: AppConfig, am: ActorRef) extends
 
     //todo:wont work with non uniform containers
     (0 until requestCount).foreach(request => {
-      am ! ContainerRequestMessage(containerCount, appConfig.getEnv(GEARPUMPMASTER_VCORES).toInt)
+      resourceManagerClient ! AdditionalContainersRequest(containerCount)
```

Comment:
@clockfly Yes I like that logic since we would be separating Callback messages from YarnAppMaster messages. However I suggest we do this in #710 since it will also incorporate FSM. Note that not all states are being tracked right now - for example in NodeManagerCallback, many of the callback methods are stubbed and do not propagate state forward to anyone. I think getting state right in #710 will take some work but should incorporate the above suggestion.